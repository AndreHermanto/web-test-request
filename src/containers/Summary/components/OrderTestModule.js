import React, { Component } from 'react';
import { 
  Row, 
  Col,
  Glyphicon,
  Button
} from 'react-bootstrap';
import { 
  SummaryBox,
  SummaryHeading,
  SummaryBreakLine,
  SummaryDetails,
  Gene,
  GeneLabel
} from '../summaryStyled'; 
/**
* OrderTestModule summary - UI for summary page to display all genes data.
*/
class OrderTestModule extends Component
{
  constructor(props) {
    super(props);
    this.handleOnClick = this.handleOnClick.bind(this);
  }
  handleOnClick() {
    this.props.handleOnClick('step1');
    return 1;
  }
  render() {
    return (
      <SummaryBox>
        <SummaryHeading> 
          Test ordered 
          {this.props.handleOnClick &&
            <Button 
              bsStyle="link"
              onClick={this.handleDetailsClick}
            >
              <Glyphicon glyph="pencil"/> Edit
            </Button>
          }
        </SummaryHeading>
        <SummaryBreakLine/>
        <Row>
          <Col md={12}>
              <SummaryDetails> <strong>{this.props.orderTestModule.test.label}</strong></SummaryDetails>
          </Col>
          <Col md={12}>
            {
              (this.props.orderTestModule.test.geneLists.length > 0) && 
              <SummaryDetails style={{ textTransform: 'capitalize', fontWeight: 400 }}> 
                {this.props.orderTestModule.test.geneLists[0].type} panel - included genes: 
                <Gene>
                  {this.props.orderTestModule.test.geneLists[0].genes.length} 
                </Gene> 
              </SummaryDetails>
            }
            {
              (this.props.orderTestModule.test.geneLists.length > 0) && (
              <div>
              {
                this.props.orderTestModule.test.geneLists[0].genes.map((gene, $index) => {
                  return <GeneLabel key={$index}>{gene}</GeneLabel> 
                })
              }
              </div>
            )}
          </Col>
        </Row>
      </SummaryBox>
    )
  }
}

export default OrderTestModule;