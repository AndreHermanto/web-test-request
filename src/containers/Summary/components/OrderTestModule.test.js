import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import TestUtils from 'react-addons-test-utils';
import OrderTestModule from './OrderTestModule';

function getStep(value) {
  return value;
}
describe('OrderTestModule test', () => {
  const testData = {
    test: {
    id:'7052d137-a166-48b0-a52e-e05a167bd176',
    label: 'Consequatur adipisci modi laudantium tenetur ea exercitationem id',
    genes:["GZOW","KIHG","DRCO","DWBY","ZMZY","BMBP","OLZO","PZOA"]
    },
    genes:["GZOW","KIHG","DRCO","DWBY","ZMZY","BMBP","OLZO","PZOA"]
  }
  const props = {
    orderTestModule: testData,
    handleOnClick: getStep
  };
    
  test('renders without crashing ', () => {
    const page = renderer.create(React.createElement(OrderTestModule, props)).toJSON();
    expect(page).toMatchSnapshot();
  });
  
  test('handleOnClick test', () => {
    var page = TestUtils.renderIntoDocument(
      <OrderTestModule orderTestModule={testData} handleOnClick={getStep}/>
    );
    expect(page.props.handleOnClick('step1')).toEqual('step1');
    expect(page.handleOnClick()).toEqual(1);
  });
});
