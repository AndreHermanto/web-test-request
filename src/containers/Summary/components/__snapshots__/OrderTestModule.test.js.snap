exports[`OrderTestModule test renders without crashing  1`] = `
<div
  className="iKCvIz">
  <h3
    className="iArjKn">
    Test ordered
    <button
      className="btn btn-link"
      disabled={false}
      onClick={undefined}
      type="button">
      <span
        className="glyphicon glyphicon-pencil" />
       Edit
    </button>
  </h3>
  <div
    className="dgEelt" />
  <div
    className="row">
    <div
      className="col-md-12">
      <p
        className="cEiVmq">
         
        <strong>
          Consequatur adipisci modi laudantium tenetur ea exercitationem id
        </strong>
      </p>
    </div>
    <div
      className="col-md-12">
      <p
        className="cEiVmq"
        style={
          Object {
            "fontWeight": 400,
            "textTransform": "capitalize",
          }
        }>
        complete
         panel - included genes:
        <span
          className="creeDZ label label-default">
          8
        </span>
      </p>
      <div>
        <span
          className="hqFlMa label label-default">
          GZOW
        </span>
        <span
          className="hqFlMa label label-default">
          KIHG
        </span>
        <span
          className="hqFlMa label label-default">
          DRCO
        </span>
        <span
          className="hqFlMa label label-default">
          DWBY
        </span>
        <span
          className="hqFlMa label label-default">
          ZMZY
        </span>
        <span
          className="hqFlMa label label-default">
          BMBP
        </span>
        <span
          className="hqFlMa label label-default">
          OLZO
        </span>
        <span
          className="hqFlMa label label-default">
          PZOA
        </span>
      </div>
    </div>
  </div>
</div>
`;
